src/protocol.h

```diff
@@ -52,47 +60,36 @@ class CMessageHeader
              READWRITE(nChecksum);
         }
 
-    // TODO: make private (improves encapsulation)
-	//HALFORD: 12-26-2014 - Add Encryption to messages - Increase size by 32 for checksum + delimiters + 10 for timestamp = 50 = 62 (vs 12)
-    public:
-        enum {
-            MESSAGE_START_SIZE=sizeof(::pchMessageStart),
-            COMMAND_SIZE=12,
-            MESSAGE_SIZE_SIZE=sizeof(int),
-            CHECKSUM_SIZE=sizeof(int),
-
-            MESSAGE_SIZE_OFFSET=MESSAGE_START_SIZE+COMMAND_SIZE,
-            CHECKSUM_OFFSET=MESSAGE_SIZE_OFFSET+MESSAGE_SIZE_SIZE,
-            HEADER_SIZE=MESSAGE_START_SIZE+COMMAND_SIZE+MESSAGE_SIZE_SIZE+CHECKSUM_SIZE
-        };
         char pchMessageStart[MESSAGE_START_SIZE];
         char pchCommand[COMMAND_SIZE];
-        unsigned int nMessageSize;
+        uint32_t nMessageSize;
         unsigned int nChecksum;
 };
 
 /** nServices flags */
-enum
-{
+enum ServiceFlags : uint64_t {
+    // NODE_NETWORK means that the node is capable of serving the complete block chain.
     NODE_NETWORK = (1 << 0),
 };
 
 /** A CService with information about it as peer */
 class CAddress : public CService
 {
-    public:
-        CAddress();
-        explicit CAddress(CService ipIn, uint64_t nServicesIn=NODE_NETWORK);
+    static constexpr uint32_t TIME_INIT{100000000};
 
-        void Init();
+    public:
+        CAddress() : CService{} {};
+        explicit CAddress(CService ipIn, ServiceFlags nServicesIn=NODE_NETWORK): CService{ipIn}, nServices{nServicesIn} {};
 
         ADD_SERIALIZE_METHODS;
 
         template <typename Stream, typename Operation>
         inline void SerializationOp(Stream& s, Operation ser_action)
         {
             if (ser_action.ForRead()) {
-                Init();
+                nServices = NODE_NETWORK;
+                nTime = 100000000;
+                nLastTry = 0;
```

Comment:
Since you got this far, I'd go ahead and make the protocol.cpp file and put `Init()` back.

Reply:
I actually have more changes locally. Almost all changes from Bitcoins protocol.h, but since it also needed changes to serialization.h, main.cpp, I don't know if it's not going too far.